test/extensions/key_value/file_based/key_value_store_test.cc

```diff
@@ -34,42 +37,74 @@ class KeyValueStoreTest : public testing::Test {
   std::string filename_;
   std::unique_ptr<FileBasedKeyValueStore> store_{};
   std::chrono::seconds flush_interval_{5};
+  Event::MockTimer* ttl_timer_ = nullptr;
   Event::MockTimer* flush_timer_ = nullptr;
+  Event::SimulatedTimeSystem test_time_;
 };
 
 TEST_F(KeyValueStoreTest, Basic) {
   EXPECT_EQ(absl::nullopt, store_->get("foo"));
-  store_->addOrUpdate("foo", "bar");
+  store_->addOrUpdate("foo", "bar", absl::nullopt);
   EXPECT_EQ("bar", store_->get("foo").value());
-  store_->addOrUpdate("foo", "eep");
+  store_->addOrUpdate("foo", "eep", absl::nullopt);
   EXPECT_EQ("eep", store_->get("foo").value());
   store_->remove("foo");
   EXPECT_EQ(absl::nullopt, store_->get("foo"));
 }
 
+TEST_F(KeyValueStoreTest, TTL) {
+  test_time_.setSystemTime(std::chrono::milliseconds(0));
+  EXPECT_CALL(*ttl_timer_, enableTimer(std::chrono::milliseconds(5000), _));
+  store_->addOrUpdate("foo", "bar", std::chrono::seconds(5));
+  EXPECT_EQ("bar", store_->get("foo").value());
+  // Advance timer to trigger expiry
+  test_time_.setSystemTime(std::chrono::milliseconds(10000));
+  ttl_timer_->invokeCallback();
+  EXPECT_EQ(absl::nullopt, store_->get("foo"));
+}
+
+TEST_F(KeyValueStoreTest, TTLUpdate) {
+  test_time_.setSystemTime(std::chrono::milliseconds(0));
+
+  store_->addOrUpdate("foo", "bar", std::chrono::seconds(10));
```

Comment:
Please add the EXPECT_CALLs on the timer here. And while we're at it, let's advance the time so we get:
t=0: add(ttl=10)
t=1: add(ttl=5)
t=6: timeFires

The important thing about the EXPECT_CALL is that it guarantees that the expiration is scheduled at the correct time.